DeliveryCompany.AppForDrivers/PackageServices.cs

```diff
@@ -0,0 +1,166 @@
+﻿using DeliveryCompany.AppForDrivers.Models;
+using DeliveryCompany.AppForDrivers.SpaceTimeProviders;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+
+namespace DeliveryCompany.AppForDrivers
+{
+    public interface IPackageServices
+    {
+        DateTime GetStartTime();
+        void MarkDeliveredPackage(List<Package> waybill, List<WaybillItem> waybillItems);
+    }
+
+    public class PackageServices : IPackageServices
+    {
+        private readonly ITimeProvider _fastForwardTimeProvider;
+        private readonly IIoHelper _ioHelper;
+
+        public PackageServices(
+            ITimeProvider fastForwardTimeProvider,
+            IIoHelper ioHelper)
+        {
+            _fastForwardTimeProvider = fastForwardTimeProvider;
+            _ioHelper = ioHelper;
+        }
+
+        public DateTime GetStartTime()
+        {
+            if (0 < _fastForwardTimeProvider.Now.Hour && _fastForwardTimeProvider.Now.Hour < 8)
+            {
+                return _fastForwardTimeProvider.Now;
+            }
+            else
+            {
+                return _fastForwardTimeProvider.Now.Date.AddHours(8);
+            }
+        }
+
+        public void MarkDeliveredPackage(List<Package> waybill, List<WaybillItem> waybillItems)
+        {
+            var deliveredPackageId = _ioHelper.GetIntFromUser("Provide the number of the delivered package:");
+            if (!waybill.Any(x => x.Id == deliveredPackageId))
+            {
+                Console.WriteLine("There is no such package.");
+                return;
+            }
+            if (waybill.Any(x => x.Id == deliveredPackageId && x.State == StateOfPackage.DeliveredManually))
+            {
+                Console.WriteLine("Package has already been delivered.");
+                return;
+            }
+            if (waybill.Any(x => x.Id == deliveredPackageId && x.ModeWaybill == ModeOfWaybill.Automatic))
+            {
+                CheckPackage(deliveredPackageId, waybill, waybillItems);
+                return;
+            }
+            else
+            {
+                UpdatePackageStatus(deliveredPackageId, waybill, waybillItems);
+                return;
+            }
+        }
+
+        private void CheckPackage(int deliveredPackageId, List<Package> waybill, List<WaybillItem> waybillItems)
+        {
+            var package = GetPackage(deliveredPackageId);
+            if (package == null)
+            {
+                Console.WriteLine("There is no such package.");
+                return;
+            }
+            if (package.State == StateOfPackage.Received)
+            {
+                Console.WriteLine("Package has already been delivered.");
+                return;
+            }
+            else
+            {
+                UpdatePackageStatus(deliveredPackageId, waybill, waybillItems);
```

Comment:
This violates single method responsibility - method that says Checks something should not Update the data. Consider changing dataflow here.